toxcore/DHT.c

```diff
@@ -198,6 +198,27 @@ int to_host_family(IP *ip)
     }
 }
 
+int write_dht_packet(uint8_t *target, uint16_t target_size, const uint8_t pkt_type,
+                     const uint8_t pub_key[crypto_box_PUBLICKEYBYTES],
+                     const uint8_t nonce[crypto_box_NONCEBYTES], const uint8_t *data,
+                     const uint16_t data_size) {
+  if (!target) {
+    return -1;
+  }
+
+  if (target_size - 1 - crypto_box_PUBLICKEYBYTES - crypto_box_NONCEBYTES < data_size) {
```

Comment:
I actually had that check rolled into the  the null check above. I ended up removing it because the calling function needs to also provide the type ( the -1 ), the  pubkey, and the nonce. If you're providing a buffer smaller than the data you're providing, you get what you deserve?

Thoughts?